### PR TITLE
Update docstrings and annotations in formulary length calculations

### PR DESCRIPTION
**Pull request by namurphy**
## Description

I'm updating the docstring of `gyroradius`, including some reorganization and more narrative examples.

## Motivation and context

I was looking through the online documentation and noticed that some of the symbols in the docstring for `gyroradius` were specific to ions.

### CODE REVIEW

File: plasmapy/formulary/lengths.py [hunk 1]
```diff
@@ -101,18 +106,19 @@ def Debye_length(T_e: u.K, n_e: u.m**-3) -> u.m:
...
 @particle_input(any_of={"charged", "uncharged"})
 def gyroradius(  # noqa: C901
-    B: u.T,
+    B: u.Quantity[u.T],
     particle: ParticleLike,
     *,
-    Vperp: u.m / u.s = np.nan * u.m / u.s,
-    T: u.K = None,
+    Vperp: u.Quantity[u.m / u.s] = np.nan * u.m / u.s,
+    T: u.Quantity[u.K] = None,
```
Comment (namurphy) [inline plasmapy/formulary/lengths.py:113]: Excited that these `Quantity` annotations are working!  Now these can be true type hints.

File: plasmapy/diagnostics/thomson.py [hunk 2]
```diff
@@ -653,7 +653,7 @@ def _spectral_density_model(wavelengths, settings=None, **params):
     electron_vel = electron_speed[:, np.newaxis] * electron_vdir
     ion_vel = ion_speed[:, np.newaxis] * ion_vdir
 
-    # Convert temperatures from eV to Kelvin (required by fast_spectral_density)
+    # Convert temperatures from eV to kelvin (required by fast_spectral_density)
```
Comment (ejohnson-96) [inline plasmapy/diagnostics/thomson.py:656]: Wasn't it correct capitalised?

File: plasmapy/formulary/lengths.py [hunk 3]
```diff
@@ -91,7 +96,7 @@ def Debye_length(T_e: u.K, n_e: u.m**-3) -> u.m:
 
 
 @validate_quantities(
-    Vperp={"can_be_nan": True},
+    Vperp={"can_be_nan": True},  # none_shall_pass
```
Comment (ejohnson-96) [inline plasmapy/formulary/lengths.py:99]: 👍

File: plasmapy/formulary/lengths.py (hunk 1 above)
Comment (ejohnson-96) [reply plasmapy/formulary/lengths.py:113]: Will these have to be altered in all other modules?

**Review by ejohnson-96 — APPROVED** [review]
Looks good

File: plasmapy/diagnostics/thomson.py (hunk 2 above)
Comment (pheuer) [reply plasmapy/diagnostics/thomson.py:656]: I thought so too, but apparently not: https://en.wikipedia.org/wiki/Kelvin#:~:text=The%20word%20%22kelvin%22%20is%20not,K%20is%20a%20capital%20letter.

File: plasmapy/formulary/lengths.py (hunk 1 above)
Comment (namurphy) [reply plasmapy/formulary/lengths.py:113]: Yep!  This is a first try to see if the new capability of `@validate_quantities` to deal with `Quantity[Unit]` style is working in action.  I'm hoping to figure out a way to automate this...which might end up meaning I'll need to delve into regular expressions.

**Pull request merged**